### ISSUE TITLE
Improve RESOURCE_EXHAUSTED response message

### ISSUE THREAD

**Issue by korthout** [camunda/camunda#8660]
**Is your feature request related to a problem? Please describe.**
When a broker is responding with `RESOURCE_EXHAUSTED`, the client will receive the following message:

>Expected to execute the command on one of the partitions, but all failed; there are no more partitions available to retry. Please try again. If the error persists contact your zeebe operator

Even though this message says to retry, to users it may seem that even a single instance of this error means something went wrong. This often leads to questions in our community (forums, slack, etc) as well as too support cases.

Whether it really is a problematic situation for the client depends on the reason (e.g. OOD is bad while append-backpressure could be a temporary hick-up; note that the reason is currently lost in the gateway https://github.com/camunda-cloud/zeebe/issues/6055) and the frequency of the `RESOURCE_EXHAUSTED` response (i.e. are the broker(s) being hammered with requests or was there a short moment of pressure that was immediately resolved?). 

**Describe the solution you'd like**
Improve the wording of the `RESOURCE_EXHAUSTED` response message, to make it clear to users what happened, what the impact is and what they should do about it. Important is to make it clear that there exists a difference between a real problem, and a few occurrences. 

It's also important to highlight how users should handle these errors: i.e. apply a back-off strategy.

Finally, we could consider to refer to our docs on [handling technical errors](https://docs.camunda.io/docs/apis-clients/grpc/#technical-error-handling), although both versioning and link-decay make this complicated.

**Describe alternatives you've considered**
🤔 We could improve our troubleshooting docs, to make it easier for users to find a solution when they encounter this problem.

**Additional context**
- related the reason for the `RESOURCE_EXHAUSTED` is currently lost in the gateway https://github.com/camunda-cloud/zeebe/issues/6055)

**Comment by npepinpe** [comment]
I don't think this fits with any of our prioritized goals for this quarter at the moment, unfortunately, so I will pass this on to our product team. However, if anyone has a more concrete proposal, then I'm happy to evaluate with them if this is something we can achieve this quarter on top of the current product goals, and if so we could go ahead and do it.
